### ISSUE TITLE
binary distribution

### ISSUE THREAD

**Issue by sylware** [nwjs/nw.js#7804]
I was made aware about this software via the game "echoed memories" on steam.

It does not run on lean distros (or Qt only desktop, x11 only):
The engine should be linked with '-static-libgcc -static-libstdc++' gcc options (or clang equilavent) to prevent long term ABI issues.
Only expect x11 or wayland, not Qt, nor GTK+ (or wxwidget/motif/etc). 
All "third-party" libs should be statically linked in the engine and those which cannot (symbol conflicts or dynamically loading is core design) should be distributed as shared libs (and dynamically loaded).
For instance libcrypto.so.1.1 (binary apps should be able to hardcode their ca bundle into themselves if they need some, with some gracefull fallback of ca bundle locations).
Dynamically load the core system libs (x11, libasound, wayland code should be static in the engine except libxkbcommon which should by dynamically loaded). To work around the planned obsolescence from GNU symbol versioning, even symbols from the glibc libs should be resolved (except dlopen,dlsym and dlclose from libdl ofc).
Provide a graceful fallback or renderers on x11/wayland: vulkan->GL->CPU.

Basically, a binary app should statically load only libdl and libc (in ELF DT_NEEDED).

Thank you for listening to this.
Hope this help improve the compatibility across glibc/linux distros of the binary apps based on your engine.

**Comment by sylware** [comment]
news update: as of today, gcc libstc++ and libgcc_s are still NOT fit for binary -only distributions.

You could switch to libdl mode easily with plain C, does not seem reasonable with c++ unless the gcc devs fix their code.

My negative opinion of c++ worsen again.
